### PR TITLE
Chore: Update examples to no longer use cy.server or cy.route

### PR DESCRIPTION
**Pull request by mjhenkes**
cy.server and cy.route are deprecated and removed in cypress 12.

### CODE REVIEW

File: examples/logging-in__single-sign-on/cypress/e2e/logging-in-single-sign-on-spec.cy.js [hunk 1]
```diff
@@ -240,9 +239,11 @@ describe('Logging In - Single Sign on', function () {
           // note again this test uses "function () { ... }" callback
           // in order to get access to the test context "this.token" saved above
 
-          cy.server()
           // observe the "GET /config" call from the application
-          cy.route('/config').as('getConfig')
+          cy.intercept('/config', (req) => {
+            // a work around to prevent cached results
```
Comment (mjhenkes) [inline examples/logging-in__single-sign-on/cypress/e2e/logging-in-single-sign-on-spec.cy.js:244]: This works around 304's not returning the response body.

File: examples/logging-in__xhr-web-forms/cypress/e2e/logging-in-xhr-web-form-spec.cy.js [hunk 2]
```diff
@@ -131,18 +131,18 @@ describe('Logging In - XHR Web Form', function () {
...
-      cy.route({
-        method: 'POST',
-        url: '/login',
-        response: {
+      cy.intercept(
+        'POST',
+        '/login',
+        {
+          body: {
           // simulate a redirect to another page
```
Comment (BlueWinds) [inline examples/logging-in__xhr-web-forms/cypress/e2e/logging-in-xhr-web-form-spec.cy.js:141]:
```suggestion
            // simulate a redirect to another page
```

File: examples/server-communication__xhr-assertions/cypress/e2e/multiple-requests.cy.js [hunk 3]
```diff
@@ -19,22 +18,22 @@ describe('waits', () => {
...
+      expect(xhr1.response, 'first request').to.have.property('statusCode', 201)
     })
 
     // we cannot guarantee the order of XHR requests - they really depend on the
     // server response speed. Sometimes a later request finishes first.
     // all we can say that each request should receive a response with
     // id equal to 101 or 102
-    cy.get('@post.1').its('response.body.id').should('be.oneOf', [101, 102])
-    cy.get('@post.2').its('response.body.id').should('be.oneOf', [101, 102])
+    cy.get('@post.1').its('response').its('body.id').should('be.oneOf', [101, 102])
```
Comment (BlueWinds) [inline examples/server-communication__xhr-assertions/cypress/e2e/multiple-requests.cy.js:36]: Can these be `.its('response.body.id')`?

Comment (mjhenkes) [reply examples/server-communication__xhr-assertions/cypress/e2e/multiple-requests.cy.js:36]: They sure can.

File: package.json [hunk 4]
```diff
@@ -106,7 +106,7 @@
     "common-tags": "1.8.0",
     "console.table": "0.10.0",
     "cy-spok": "1.3.2",
-    "cypress": "10.7.0",
+    "cypress": "https://cdn.cypress.io/beta/npm/12.0.0/linux-x64/release/12.0.0-d7e148f8cde4f74300bec3a3e2fa55ed3139d6fd/cypress.tgz",
```
[the remark this answers is not in the document]
Comment (BlueWinds) [reply package.json:109]: Will we need to go back and update this after release, and is there a task down written somewhere for it?

**Review by BlueWinds — APPROVED** [review]

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (mjhenkes) [reply package.json:109]: Yeah, we will need to go back and update this. It should be done as part of the release process: https://github.com/cypress-io/cypress/blob/develop/guides/release-process.md?plain=1#L187

**Review by mschile — APPROVED** [review]

**Pull request merged**